File: solutions_5686275109552128_1/Python/roiti46/large.py
T = int(raw_input())
for loop in xrange(T):
    D = int(raw_input())
    P = map(int, raw_input().split())
    ans = max(P)
    P = [[p,1] for p in sorted(P)]

    for divide in xrange(1,1001):
        P[-1][1] += 1
        P = sorted(P, key = lambda x: x[0]/x[1] + (1 if x[0]%x[1] else 0))
        tmp = max(p/eat + (1 if p%eat else 0) for p ,eat in P) + divide
        ans = min(tmp, ans)
        
    print "Case #%d: %d"%(loop+1, ans)

File: solutions_5686275109552128_1/C++/AKM/large.cpp
#include <iostream>

using namespace std;

const int MAXN = 1010;

int a[MAXN];

int main()
{
	int tt;
	cin >> tt;
	for (int t = 1; t <= tt; t++)
	{
		int n;
		cin >> n;
		int cmx = 0;
		for (int i = 0; i < n; i++)
		{
			cin >> a[i];
			cmx = max(cmx, a[i]);
		}
		int ans = cmx;
		for (int mx = 1; mx <= cmx; mx++)
		{
			int add = 0;
			for (int i = 0; i < n; i++)
				add += (a[i] - 1) / mx;
			ans = min(ans, add + mx);
		}
		cout << "Case #" << t << ": " << ans << endl;
	}
}
